common/src/main/java/dev/ftb/mods/ftbchunks/data/TeamMemberData.java

```diff
@@ -0,0 +1,130 @@
+import net.minecraft.core.Registry;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.Tag;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.level.Level;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class TeamMemberData {
+    private int maxClaims;
+    private int maxForceLoads;
+    private boolean offlineForceLoader;
+    private final Set<ChunkDimPos> originalClaims;
+
+    private TeamMemberData(int maxClaims, int maxForceLoads, boolean offlineForceLoader, Set<ChunkDimPos> originalClaims) {
+        this.maxClaims = maxClaims;
+        this.maxForceLoads = maxForceLoads;
+        this.offlineForceLoader = offlineForceLoader;
+        this.originalClaims = originalClaims;
+    }
+
+    public static TeamMemberData defaultData() {
+        return new TeamMemberData(FTBChunksWorldConfig.MAX_CLAIMED_CHUNKS.get(), FTBChunksWorldConfig.MAX_FORCE_LOADED_CHUNKS.get(), false, new HashSet<>());
+    }
+
+    public static TeamMemberData deserializeNBT(CompoundTag tag) {
+        int maxClaims = tag.getInt("max_claimed_chunks");
+        int maxForced = tag.getInt("max_force_loaded_chunks");
+        boolean offline = tag.getBoolean("offline_force_loader");
+        Set<ChunkDimPos> orig = readOriginalClaims(tag.getCompound("original_claims"));
+
+        return new TeamMemberData(maxClaims, maxForced, offline, orig);
+    }
+
+    public static TeamMemberData fromPlayerData(ServerPlayer player, FTBChunksTeamData otherTeam) {
+        return new TeamMemberData(
+                otherTeam.getMaxClaimChunks() + otherTeam.extraClaimChunks,
+                otherTeam.getMaxForceLoadChunks() + otherTeam.extraForceLoadChunks,
+                FTBChunksWorldConfig.canPlayerOfflineForceload(player),
+                new HashSet<>(otherTeam.getClaimedChunks().stream().map(cc -> cc.pos).toList())
+        );
+    }
+
+    public CompoundTag serializeNBT() {
+        CompoundTag tag = new CompoundTag();
+
+        tag.putInt("max_claimed_chunks", maxClaims);
+        tag.putInt("max_force_loaded_chunks", maxForceLoads);
+        tag.putBoolean("offline_force_loader", offlineForceLoader);
+        CompoundTag origTag = writeOriginalClaims();
+        if (!origTag.isEmpty()) tag.put("original_claims", origTag);
+
+        return tag;
+    }
+
+    private CompoundTag writeOriginalClaims() {
+        CompoundTag origTag = new CompoundTag();
+
+        Map<String, ListTag> perDimensionTags = new HashMap<>();
+        originalClaims.forEach(cdp -> {
+            ListTag l = perDimensionTags.computeIfAbsent(cdp.dimension.location().toString(), k -> new ListTag());
+            SNBTCompoundTag cdpTag = new SNBTCompoundTag();
+            cdpTag.singleLine();
+            cdpTag.putInt("x", cdp.x);
```

Comment:
to / from long? Minor thing, just noting things I see as I go :P

Reply:
Would be better, yes - but migration code would also be needed to handle existing saves, which is extra complexity.  Might be better to wait till a big 1.20 refactor?